simplq/src/main/java/me/simplq/service/TokenService.java

```diff
@@ -57,6 +58,11 @@ public TokenDetailResponse getToken(String queueId, String contactNumber) {
   @Transactional
   public TokenDeleteResponse deleteToken(String tokenId) {
     tokenRepository.setTokenStatusById(TokenStatus.REMOVED, tokenId);
+    /*Set Token Deletion timestamp to now*/
+    Date nowTime = new Date();
+
+    tokenRepository.findById(tokenId).get().setTokenDelete( nowTime);
+
     return tokenRepository
```

Comment:
Instead of doing the double retrieve from database, you can reuse `tokenRepository.findById(tokenId)` using a local variable.